src/app/clusters/time-synchronization-server/time-synchronization-cluster.h

```diff
@@ -69,48 +65,57 @@ enum class TimeSyncEventFlag : uint8_t
     kMissingTTSource = 16,
 };
 
-void SetDefaultDelegate(Delegate * delegate);
-Delegate * GetDefaultDelegate();
+} // namespace TimeSynchronization
 
-class TimeSynchronizationServer : public FabricTable::Delegate
+class TimeSynchronizationCluster : public DefaultServerCluster,
+                                   public FabricTable::Delegate
 #if TIME_SYNC_ENABLE_TSC_FEATURE
     ,
-                                  public ReadClient::Callback
+                                   public ReadClient::Callback
 #endif
 {
 public:
-    TimeSynchronizationServer();
-    void Init();
-    void Shutdown();
-
-    static TimeSynchronizationServer & Instance(void);
-    TimeSyncDataProvider & GetDataProvider(void) { return mTimeSyncDataProvider; }
-
-    CHIP_ERROR SetTrustedTimeSource(const DataModel::Nullable<Structs::TrustedTimeSourceStruct::Type> & tts);
-    CHIP_ERROR SetDefaultNTP(const DataModel::Nullable<chip::CharSpan> & dntp);
-    void InitTimeZone(void);
-    CHIP_ERROR SetTimeZone(const DataModel::DecodableList<Structs::TimeZoneStruct::Type> & tzL);
-    CHIP_ERROR LoadTimeZone(void);
-    CHIP_ERROR ClearTimeZone(void);
-    void InitDSTOffset(void);
-    CHIP_ERROR SetDSTOffset(const DataModel::DecodableList<Structs::DSTOffsetStruct::Type> & dstL);
-    CHIP_ERROR LoadDSTOffset(void);
-    CHIP_ERROR ClearDSTOffset(void);
-    DataModel::Nullable<Structs::TrustedTimeSourceStruct::Type> & GetTrustedTimeSource(void);
-    Span<TimeSyncDataProvider::TimeZoneStore> & GetTimeZone(void);
-    DataModel::List<Structs::DSTOffsetStruct::Type> & GetDSTOffset(void);
+    // NOTE: this set is smaller than the full optional attributes supported by time synchronization
+    //       as other attributes are controlled by feature flags
+    using OptionalAttributeSet = app::OptionalAttributeSet<TimeSynchronization::Attributes::TimeSource::Id>;
+
+    TimeSynchronizationCluster(EndpointId endpoint, const OptionalAttributeSet & optionalAttributeSet,
+                               const BitFlags<TimeSynchronization::Feature> features,
+                               TimeSynchronization::Attributes::SupportsDNSResolve::TypeInfo::Type supportsDNSResolve,
+                               TimeSynchronization::TimeZoneDatabaseEnum timeZoneDatabase,
+                               TimeSynchronization::TimeSourceEnum timeSource,
+                               TimeSynchronization::Attributes::NTPServerAvailable::TypeInfo::Type ntpServerAvailable);
+
+    CHIP_ERROR Startup(ServerClusterContext & context) override;
+    void Shutdown() override;
+
+    TimeSyncDataProvider & GetDataProvider() { return mTimeSyncDataProvider; }
+
+    CHIP_ERROR SetTrustedTimeSource(const DataModel::Nullable<TimeSynchronization::Structs::TrustedTimeSourceStruct::Type> & tts);
+    CHIP_ERROR SetDefaultNTP(const DataModel::Nullable<CharSpan> & dntp);
+    void InitTimeZone();
```

Comment:
Do we need all these to be public? Could we reduce the API surface and make more of these private?